### PR TITLE
node:fs/promises.cp should make path if dest does not exist

### PR DESCRIPTION
**Pull request by nektro**
Closes https://github.com/oven-sh/bun/issues/11260

### CODE REVIEW

File: src/bun.js/node/node_fs.zig [hunk 1]
```diff
@@ -6627,7 +6627,12 @@ pub const NodeFS = struct {
                 mode_ |= C.darwin.COPYFILE_EXCL;
             }
 
-            return ret.errnoSysP(C.copyfile(src, dest, null, mode_), .copyfile, src) orelse ret.success;
+            const first_try = ret.errnoSysP(C.copyfile(src, dest, null, mode_), .copyfile, src) orelse return ret.success;
```
Comment (Jarred-Sumner) [inline src/bun.js/node/node_fs.zig:6630]: What about linux and Windows?

File: test/js/node/fs/fs.test.ts [hunk 2]
```diff
@@ -3216,3 +3216,18 @@ it("promises.fdatasync with a bad fd should include that in the error thrown", a
...
+  let folder = "folder-not-exist";
+  let dst = join(folder, src);
+
+  src = join(x_dir, src);
+  folder = join(x_dir, folder);
+  dst = join(x_dir, dst);
+
+  await promises.writeFile(src, "A".repeat(131073));
+  await promises.rm(folder, { recursive: true, force: true });
+  await promises.cp(src, dst);
```
Comment (Jarred-Sumner) [inline test/js/node/fs/fs.test.ts:3233]: Where does this test check that the `dst` is equal to `"A".repeat(131073)`?

File: src/bun.js/node/node_fs.zig (hunk 1 above)
Comment (nektro) [reply src/bun.js/node/node_fs.zig:6630]: linux is handled; will push an update with windows

File: test/js/node/fs/fs.test.ts (hunk 2 above)
Comment (nektro) [reply test/js/node/fs/fs.test.ts:3233]: will add this

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**